Cabal/src/Distribution/Simple/Setup.hs

```diff
@@ -556,6 +564,17 @@ configureOptions showOrParseArgs =
                 "Don't emit debug info"
          ]
 
+      , multiOption "dump-buildinfo"
```

Comment:
Why conditionally? Is there a reason not to write it always?

Reply:
I did not want to change the default behaviour, but it would be indeed better if it was always generated. Subject to bikeshed.

Reply:
performance is not a problem I suppose

Reply:
I woudn't be so sure that performance couldn't be a problem, best to err on the side of caution when people with pitchforks are sure to follow if `build` speed regresses ;)

If it turns out to be unproblematic it's an easy default to change: just ignore the flag going forward, but the other way around isn't so easy since tooling would break if we introduce a flag.

Lastly my understanding is that hie-bios currently always does at least one (implicit) build call anyway though the repl command so we're just maintaining the status quo if it has to do one with a special flag to get the buildinfo.

Reply:
> Lastly my understanding is that hie-bios currently always does at least one (implicit) build call anyway though the repl command so we're just maintaining the status quo if it has to do one with a special flag to get the buildinfo.

yeah but it would be nice if we get to make no one if possible, though